### PR TITLE
Upgrade to targetSdk 33

### PR DESCRIPTION
**Pull request by ithinkihaveacat**
Also minor change to clarify isAmbient log messages.

### CODE REVIEW

**Review by yschimke — APPROVED** [review]

**Comment by yschimke** [comment]
```
/home/runner/work/wear-os-samples/wear-os-samples/AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt:176: Error: When scheduling exact alarms, apps should explicitly call AlarmManager#canScheduleExactAlarms or handle `SecurityException`s [ScheduleExactAlarm]
                    ambientUpdateAlarmManager.setExact(
```

**Comment by ithinkihaveacat** [comment]
> /home/runner/work/wear-os-samples/wear-os-samples/AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt:176: Error: When scheduling exact alarms, apps should explicitly call AlarmManager#canScheduleExactAlarms or handle `SecurityException`s [ScheduleExactAlarm]
>                     ambientUpdateAlarmManager.setExact(

@yschimke Is `set(.)` going to be fine here, do you think? I don't see why it needs to be exact.

**Comment by ithinkihaveacat** [comment]
> > /home/runner/work/wear-os-samples/wear-os-samples/AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt:176: Error: When scheduling exact alarms, apps should explicitly call AlarmManager#canScheduleExactAlarms or handle `SecurityException`s [ScheduleExactAlarm]
> > ambientUpdateAlarmManager.setExact(
> 
> @yschimke Is `set(.)` going to be fine here, do you think? I don't see why it needs to be exact.

Ok, should've tested this, but it appears `setExact()` is need to break out of ambient mode. I added a comment instead to explain what's going on.

File: AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt [hunk 1]
```diff
@@ -173,6 +175,10 @@ fun AlwaysOnApp(
                     val triggerTime = currentInstant.getNextInstantWithInterval(
                         AMBIENT_INTERVAL
                     )
+                    // setExact() should *not* be used in production code. In a real application,
```
Comment (ithinkihaveacat) [inline AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt:178]: @garanj @yschimke Is this accurate?

Comment (yschimke) [reply AlwaysOnKotlin/compose/src/main/java/com/example/android/wearable/wear/alwayson/AlwaysOnApp.kt:178]: Lets not show it then.  Let's remove it.

Either say
a) Leave it as the default behaviour on Wear 3 + 4
b) Explain the difference and Wear 4 doesn't allow.

We shouldn't show working around platform efficiency restrictions.

**Comment by ithinkihaveacat** [comment]
Abandoning this; after discussion we will do the bare minimum re targetSdk updates.

**Pull request closed without merging**